internal/node/node.go

```diff
@@ -32,16 +32,22 @@ func NewNode(client client.Client) Node {
 	}
 }
 
-func (n *node) IsNodeSchedulable(node *v1.Node) bool {
+func (n *node) IsNodeSchedulable(node *v1.Node, tolerations []v1.Toleration) bool {
+TAINTLOOP:
 	for _, taint := range node.Spec.Taints {
+		for _, toleration := range tolerations {
+			if taint.Key == toleration.Key && taint.Value == toleration.Value && taint.Effect == toleration.Effect {
+				continue TAINTLOOP
+			}
+		}
 		if taint.Effect == v1.TaintEffectNoSchedule {
 			return false
 		}
 	}
 	return true
```

Comment:
```suggestion
var toleratedTaints int
for _, taint := range node.Spec.Taints {
    for _, toleration := range tolerations {
        if toleration.ToleratesTaint(taint) {
            toleratedTaints++
            break
        }
    }
}
return toleratedTaints == len(node.Spec.Taints)
```

WDYT?


@yevgeny-shnaidman this is changing the original behavior but I believe this is a more "complete" solution.

Reply:
@ybettan your code looks good, but you are missing some corner cases:
if there is no toleration exists for the taint, but taints' effect is missing or it's effect is PreferNoSchedule, then the function should return true. 
@tsprasannaa a couple of minor comments for your code:  i would prefer not to use labels (TAINTLOOP) if not really needed. In this case it can be solved by a variable in the loop. Also, you should check if taint effect NoExecute, alongside with noSchedule

Reply:
@ybettan @yevgeny-shnaidman Comments taken care. Please take a look. Thanks